packages/mui-base/src/Select/Popup/SelectPopup.tsx

```diff
@@ -0,0 +1,111 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import type { Side } from '@floating-ui/react';
+import type { BaseUIComponentProps } from '../../utils/types';
+import { useSelectRootContext } from '../Root/SelectRootContext';
+import { useSelectPositionerContext } from '../Positioner/SelectPositionerContext';
+import { commonStyleHooks } from '../utils/commonStyleHooks';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { useForkRef } from '../../utils/useForkRef';
+import type { CustomStyleHookMapping } from '../../utils/getStyleHookProps';
+import { useSelectPopup } from './useSelectPopup';
+
+const customStyleHookMapping: CustomStyleHookMapping<SelectPopup.OwnerState> = {
+  ...commonStyleHooks,
+  entering(value) {
+    return value ? { 'data-select-entering': '' } : null;
```

Comment:
I saw you renamed it in the Menu to 'data-entering' and 'data-exiting' (https://github.com/mui/base-ui/pull/638/files#diff-e31760a8ce5d872099c8c8aa212adac7e10ad678504402a0b963903373cbda26). Let's make it consistent, then.